File: JAVA/src/main/java/com/inf_112/trabalho_final/service/CorService.java
package com.inf_112.trabalho_final.service;

import com.inf_112.trabalho_final.dao.CorDao;
import com.inf_112.trabalho_final.model.Cor;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class CorService {

    private final CorDao corDao;

    public CorService(CorDao corDao) {
        this.corDao = corDao;
    }

    public Cor getById(int id) {
        return corDao.findById(id);
    }

    public List<Cor> getAll() {
        return corDao.findAll();
    }

    public void insert(Cor cor) {
        corDao.insert(cor);
    }

    public void update(Cor cor) {
        corDao.update(cor);
    }

    public void delete(int id) {
        corDao.delete(id);
    }
}

File: C++/service/CorService.cpp
#include "CorDao.h"
#include "Cor.h"
#include <vector>

class CorService {
private:
    CorDao& corDao; // Referência ao DAO

public:
    // Construtor
    CorService(CorDao& dao) : corDao(dao) {}

    // Obter Cor por ID
    Cor getById(int id) {
        return corDao.findById(id);
    }

    // Obter todas as Cores
    std::vector<Cor> getAll() {
        return corDao.findAll();
    }

    // Inserir uma nova Cor
    void insert(const Cor& cor) {
        corDao.insert(cor);
    }

    // Atualizar uma Cor existente
    void update(const Cor& cor) {
        corDao.update(cor);
    }

    // Deletar uma Cor pelo ID
    void deleteById(int id) {
        corDao.deleteById(id);
    }
};
